### PR TITLE
Modules overhaul

### PR DESCRIPTION
**Pull request by Levi-Lesches**
With new classes like `ModuleOwnerYield` in place, we can define "click-ability" by the presence or absence of such classes. This lets us overhaul much of the code, including the Flash-facing side.

Changed semantics: 
- `Module.vote()` and `Module.execute()` have been merged into `Module.click()`
- `Module.is_setup` can now be null. The three states mean: 
  - `True`: module has been set up by the user
  - `False`: module needs setup from the user
  - `None`: module never needs to be set up
- New usage: 
  - Always use `Module.click()` when interacting with another user's module
  - `Module.is_clickable()` to tell if a module should be clicked 
  - `Module.is_setup` to tell if the user put in items to set up the module
  - This distinction is handy when it comes to showing a "click" button but not a "teardown" button.

Still TODO: 
- [x] choose the guest's yield when multiple options are present
- [x] Incorporate this logic for `ModuleSetupTrade`
- [x] ~Implement `ModuleBattle`, for modules where the owner and guest can't win the same time~ (another PR)

Tests will need to be looked at, of course.

### CODE REVIEW

File: mln/models/module.py
Comment (Levi-Lesches) [inline mln/models/module.py:76]: Should be `break`s here

File: mln/models/module.py
Comment (Levi-Lesches) [inline mln/models/module.py:110]: Try to use `self.item.setup_costs` instead

File: mln/models/module.py
Comment (Levi-Lesches) [inline mln/models/module.py:131]: Maybe the dict should have None as a key with an empty list?

File: mln/models/module.py
Comment (Levi-Lesches) [inline mln/models/module.py:178]: Leading underscore

File: mln/models/module.py
Comment (lcdr) [inline mln/models/module.py]: I'd really appreciate if you didn't randomly move around methods for no reason during a PR, this is a pain to review as I can't tell whether you're just moving stuff around or actually changing some lines as well. I see no reason to move the methods around in this case in the first place, but if there is one, please do it in a separate PR in the future.

**Review by lcdr — CHANGES_REQUESTED** [review]
All in all, this looks good. There's just a few very minor nits I found, which I commented on. I haven't run this locally, but I trust your judgement that this won't break things. Before merging, I'd like to confirm though that you're run the test suite and that everything passed. (This would ideally be done by CI at some point.)

Side note: Please organize (and if necessary, edit) your commits as to help the reviewer next time. When a PR is fairly complex, I review commit-by-commit, to break down the changes and to use your commit messages as a guide. Ideally, PRs should be organized with this in mind to begin with. This means that instead of one commit adding a function and then a second commit renaming it to a name you realized fits better, you'd amend the first commit (or squash the two commits) to only make the final changes appear at review time. The same goes for when you notice that the previous commit doesn't actually compile and add some changes to fix that. Also, when functionality is moved from one place to another, both the removal and the addition should be in the same commit, otherwise it's very confusing to tell why lines were added/removed. Rewriting commits after-the-fact is a rather advanced usage of git, but modern GUI frontends should make this fairly manageable, and it makes review much easier. Thanks!

File: mln/models/module.py
Comment (lcdr) [reply mln/models/module.py:131]: I'd be fine with that as well, I leave it up to you.

File: mln/models/module.py
Comment (lcdr) [reply mln/models/module.py:76]: Who is this addressed to? Is this resolved?

File: mln/models/module.py
Comment (Levi-Lesches) [reply mln/models/module.py]: The problem is we need to check whether the module needs setup when it is created as well since the module creation function (when a page is saved) doesn't really happen to know or care whether it needs setup. 

`Model.save` runs when a new module is created, so it serves as a migration, validation, and creation handler. Alternatively, I could split it into a migration function and override `Model.create` instead of `Module.save`, and we don't really need validation on every save.

**Pull request closed without merging**
